### PR TITLE
docs: update RapidFire AI integration with FSDP and multi-backend tracking

### PR DESCRIPTION
**Pull request by kamran-rapidfireAI**
# What does this PR do?

Refreshes the RapidFire AI integration guide ([`docs/source/rapidfire_integration.md`](https://github.com/huggingface/trl/blob/main/docs/source/rapidfire_integration.md)) to cover features shipped in `rapidfireai` v0.14.x–v0.15.x since the original guide landed in #4340 (Oct 29, 2025).

### What's new in the article

- **Multi-GPU Fully Sharded Data Parallel (FSDP)** — new subsection under "Advanced Features" with an end-to-end `RFSFTConfig` + `fsdp` / `fsdp_config` example and links to the FSDP tutorial notebooks (SFT lite, SFT large, DPO).
- **Pluggable metric tracking backends** — new subsection covering MLflow (default), TensorBoard (default in Colab), and Trackio, selectable via the new `--tracking-backends` flag or `RF_*_ENABLED` environment variables.
- **Google Colab support** — short subsection noting Colab as a zero-setup path for SFT / DPO / GRPO tutorials.
- **Reframed opener + "Why use RapidFire AI with TRL?" section** — leads with the contrast vs. plain TRL (sequential vs. concurrent training of N configs on the same GPUs) so the value prop lands on first read.

### Small accuracy fixes folded in

- Bumped PyTorch prerequisite to `2.8+` (matches current rapidfireai README) and noted that multiple GPUs are required for FSDP.
- Fixed the dashboard port (`3000` → `8853`, the default since rapidfireai v0.12.4) and normalized the dashboard URL to `http://localhost:8853` consistently.
- Updated the SFT / DPO / GRPO example notebook links to their new `tutorial_notebooks/fine-tuning/` and `tutorial_notebooks/post-training/` subfolders.
- Harmonized "clone-modify (with or without warm starting)" terminology across the article so it matches the upstream rapidfireai README.

### Scope

Documentation only — 1 file changed, +114 / -26 lines. No code changes, no new tests required. Verified that all imports, command snippets, env var defaults, and FSDP example values match `RapidFireAI/rapidfireai` v0.15.2 source.

This was reviewed internally on our fork before opening here.

## Before submitting

- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request), Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? No — this is a refresh of an existing integration doc that was previously approved and merged in #4340.
- [x] Did you make sure to update the documentation with your changes? This PR _is_ the documentation change.
- [x] Did you write any new necessary tests? n/a — docs-only change.

## AI writing disclosure

- [x] AI-assisted: some parts were suggested or improved by AI, but the PR was written and reviewed by a human.

## Who can review?

cc @qgallouedec — you reviewed and merged the original integration guide in #4340, would appreciate your eyes again. Happy to iterate on wording/structure.

---

> [!NOTE]
> **Low Risk**
> Docs-only changes; main risk is minor user confusion if any updated commands, ports, or links are incorrect.
> 
> **Overview**
> Refreshes `rapidfire_integration.md` to better frame RapidFire AI’s value prop vs plain TRL (live concurrent N-config training), adds project links, and reorganizes/expands the feature list.
> 
> Adds new documentation for **multi-GPU FSDP** (end-to-end `fsdp`/`fsdp_config` example + notebook links), **selectable experiment tracking backends** (MLflow/TensorBoard/Trackio via `--tracking-backends` or `RF_*_ENABLED` env vars), and a short **Google Colab** section.
> 
> Fixes/updates several doc details: PyTorch prerequisite to `2.8+`, clarifies multi-GPU behavior, standardizes IC Ops terminology to “clone-modify”, corrects the default dashboard URL/port to `http://localhost:8853`, and updates tutorial notebook links to new subfolders.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 3a9676d72e79e3e678f2ae42887b1397c7e9b199. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: docs/source/rapidfire_integration.md [hunk 1]
```diff
@@ -320,7 +325,90 @@ config = RFModelConfig(
...
+        target_modules=["q_proj", "k_proj", "v_proj", "o_proj"],
+        bias="none",
+    ),
+    training_args=RFSFTConfig(
+        learning_rate=2e-4,
+        per_device_train_batch_size=1,
+        gradient_accumulation_steps=8,
+        gradient_checkpointing=True,
+        gradient_checkpointing_kwargs={"use_reentrant": False},
+        bf16=True,
```
Comment (qgallouedec) [inline docs/source/rapidfire_integration.md:352]: fyi, these are default values, so you may want to remove them

File: docs/source/rapidfire_integration.md [hunk 2]
```diff
@@ -320,7 +325,90 @@ config = RFModelConfig(
...
+            "forward_prefetch": True,
+            "use_orig_params": False,
+            "cpu_ram_efficient_loading": True,
+            "offload_params": True,
+            "sync_module_states": True,
+            "limit_all_gathers": True,
+        },
+    ),
+    model_type="causal_lm",
+    model_kwargs={"torch_dtype": "auto", "use_cache": False},
```
Comment (qgallouedec) [inline docs/source/rapidfire_integration.md:367]: `"use_cache": False` isn't necessary, the forward is already run with `use_cache=False`

**Review by qgallouedec — APPROVED** [review]
lgtm, just two comments

File: docs/source/rapidfire_integration.md (hunk 1 above)
Comment (kamran-rapidfireAI) [reply docs/source/rapidfire_integration.md]: Done — removed in 3a9676d7.

File: docs/source/rapidfire_integration.md (hunk 2 above)
Comment (kamran-rapidfireAI) [reply docs/source/rapidfire_integration.md]: Done — removed in 3a9676d7.

**Comment by qgallouedec** [comment]
I'll merge this one but feel free to open other PRs if you want to address the comments
